api/src/main/java/org/pmiops/workbench/api/DiskController.java

```diff
@@ -0,0 +1,117 @@
+    this.workspaceAuthService = workspaceAuthService;
+    this.workspaceDao = workspaceDao;
+    this.leonardoMapper = leonardoMapper;
+  }
+
+  private DbWorkspace lookupWorkspace(String workspaceNamespace)
+      throws org.pmiops.workbench.exceptions.NotFoundException {
+    return workspaceDao
+        .getByNamespace(workspaceNamespace)
+        .orElseThrow(() -> new NotFoundException("Workspace not found: " + workspaceNamespace));
+  }
+
+  @Override
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
+    String googleProject = dbWorkspace.getGoogleProject();
+    try {
+      LeonardoGetPersistentDiskResponse response;
+      List<LeonardoListPersistentDiskResponse> responseList =
+          leonardoNotebooksClient.listPersistentDiskByProject(googleProject, false);
+      responseList.sort(
+          new Comparator<LeonardoListPersistentDiskResponse>() {
+            @Override
+            public int compare(
+                LeonardoListPersistentDiskResponse t1, LeonardoListPersistentDiskResponse t2) {
+              return -1 * t1.getId().compareTo(t2.getId());
+            }
+          });
+      response =
+          !responseList.isEmpty()
+              ? leonardoNotebooksClient.getPersistentDisk(
+                  googleProject, responseList.get(0).getName())
+              : null;
+
+      if (response != null && LeonardoRuntimeStatus.ERROR.equals(response.getStatus())) {
+        log.warning(
+            String.format(
+                "Observed Leonardo runtime with unexpected error status:\n%s",
+                response.getStatus()));
+      }
+
+      if (response != null) {
+        return ResponseEntity.ok(leonardoMapper.toApiDisk(response));
+
+      } else {
+        return ResponseEntity.ok(null);
+      }
+
+    } catch (NotFoundException e) {
+      return (ResponseEntity<Disk>) ResponseEntity.notFound();
+    }
+  }
+
+  @Override
+  public ResponseEntity<EmptyResponse> deleteDisk(String workspaceNamespace) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
+    String firecloudWorkspaceName = dbWorkspace.getFirecloudName();
+    workspaceAuthService.enforceWorkspaceAccessLevel(
+        workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
+    Disk disk = getDisk(workspaceNamespace).getBody();
+    if (disk != null) {
+      leonardoNotebooksClient.deletePersistentDisk(dbWorkspace.getGoogleProject(), disk.getName());
+    } else {
+      log.warning(
+          String.format(
+              "No existing persistent disk could be deleted in workspace \n%s",
+              workspaceNamespace));
+    }
+
+    return ResponseEntity.ok(new EmptyResponse());
+  }
+
+  @Override
+  public ResponseEntity<EmptyResponse> updateDisk(
+      String workspaceNamespace, String diskName, Integer diskSize) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
```

Comment:
ditto: `enforceWorkspaceAccessLevel` check?

Reply:
after reading this more carefully, I think we may not need that access check everywhere. If we are using user's credential to make the call, would Leo can do the authZ check correctly(with SAM)?
@calbach

Reply:
agree